### PR TITLE
Add ctlr semver and priority level ref eval to FlowSchemaStatus

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Recording the evaluation of the PriorityLevelConfiguration reference
in a FlowSchemaCondition was abusing the concept of "condition".
Quoting
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
: "Conditions ... are an extension mechanism intended to be used when
the details of an observation are not a priori known or would not
apply to all instances of a given Kind. For observations that are well
known and apply to all instances, a regular field is preferred."

Instead of making the Status echo the relevant Spec generation, which
is the approach discussed earlier, I chose to echo the reference in
question --- which reduces the scenarios in which a change to the
Status is needed.

Introduced Status.ControllerSemVer to enable older writers to defer to
newer ones.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
@kubernetes/sig-api-machinery-pr-reviews

**Comment by MikeSpreitzer** [comment]
/cc @lavalamp 
/cc @yue9944882 
/cc @deads2k 
@MikeSpreitzer

File: pkg/apis/flowcontrol/types.go [hunk 1]
```diff
@@ -296,6 +298,17 @@ type FlowSchemaStatus struct {
 	// +listMapKey=type
 	// +optional
 	Conditions []FlowSchemaCondition
+
+	// `controllerSemVer` holds the semantic version of the controller
+	// that set the status.  A controller of lesser semantic version
+	// is not allowed to subsequently update the status.
+	ControllerSemVer string
```
Comment (liggitt) [inline pkg/apis/flowcontrol/types.go:305]: this is fairly unusual... this means a faulty controller cannot be rolled back?

File: pkg/apis/flowcontrol/types.go [hunk 2]
```diff
@@ -296,6 +298,17 @@ type FlowSchemaStatus struct {
...
+
+	// `evaluatedPriorityLevelConfigurationReference` reports what the
+	// kube-apiservers think of the Spec's reference to a priority
```
Comment (liggitt) [inline pkg/apis/flowcontrol/types.go:308]: how does this interact with multiple kube-apiservers?

File: pkg/apis/flowcontrol/types.go [hunk 3]
```diff
@@ -318,6 +331,18 @@ type FlowSchemaCondition struct {
...
+// EvaluatedPriorityLevelConfigurationReference is a report on whether
+// a PriorityLevelConfigurationReference refers to something that
+// exists
+type EvaluatedPriorityLevelConfigurationReference struct {
+	// `reference` is the reference being reported upon
+	Reference PriorityLevelConfigurationReference
+
+	// `exists` reports whether the referenced priority level
+	// configuration object exists
+	Exists bool
```
Comment (liggitt) [inline pkg/apis/flowcontrol/types.go:343]: status fields for referential integrity status is also fairly unusual...

File: pkg/apis/flowcontrol/types.go (hunk 2 above)
Comment (MikeSpreitzer) [reply pkg/apis/flowcontrol/types.go:308]: They should all reach the same answer.
It's conceivable that if they are of different releases they might have different answers.  That's why the `ControllerSemVer` is there.

File: pkg/apis/flowcontrol/types.go (hunk 3 above)
Comment (deads2k) [reply pkg/apis/flowcontrol/types.go:343]: > status fields for referential integrity status is also fairly unusual...

While  I may choose to frame this as a condition, a referential checks seems inbounds in cases of two cluster-scoped resources managed by cluster-admins.  The question of "is it valid" answered in status seems reasonable.

Comment (lavalamp) [reply pkg/apis/flowcontrol/types.go:343]: We have "is it working yet" conditions for CRDs. I think this is extremely similar. I might phrase it as a single e.g. "operational" condition, with differing reasons depending on why it's not operational.

File: pkg/apis/flowcontrol/types.go (hunk 1 above)
Comment (MikeSpreitzer) [reply pkg/apis/flowcontrol/types.go:305]: Dang.  That's a fatal flaw.
We considered introducing an elected leader to make the one true setting, but did not agree that this is a good idea either.
See #90541 for an alternative approach: just let disagreeing kube-apiservers fight it out, but not so fast that it makes a big stink.

**Comment by MikeSpreitzer** [comment]
This is not going to work because it does not support rollback.

**Pull request closed without merging**